### PR TITLE
crc lib plus 0x83 poly for ota crc

### PR DESCRIPTION
**Pull request by AlessandroAU**

### CODE REVIEW

File: src/lib/CRC/crc.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#include "crc.h"
```
Comment (cruwaller) [inline src/lib/CRC/crc.cpp:1]: Generic. Please harmonize datatypes and use either main types or fixed width type (recommended).

**Review by cruwaller — CHANGES_REQUESTED** [review]

File: src/src/rx_main.cpp [hunk 2]
```diff
@@ -40,7 +41,7 @@ SX1280Driver Radio;
 #define DEBUG_SUPPRESS // supresses debug messages on uart
 
 hwTimer hwTimer;
-
+GENERIC_CRC8 ota_crc(ELRS_CRC_POLY);
```
[the remark this answers is not in the document]
Comment (AlessandroAU) [reply src/src/rx_main.cpp:44]: Elrs crc and crsf crc are going to use different poly after this merge so I don't think this is possible as the precomputed tables will differ.

[the remark this answers is not in the document]
Comment (cruwaller) [reply src/src/rx_main.cpp:44]: I that case you need two.
Please, introduce 2 defines and comment. Then it is easier to follow.

[the remark this answers is not in the document]
Comment (AlessandroAU) [reply src/src/rx_main.cpp:44]: I put the defines in common, you think better in rx_main/tx_main?

[the remark this answers is not in the document]
Comment (cruwaller) [reply src/src/rx_main.cpp:44]: I think common is ok for RX/TX. I would put CRSF into CRSF.h and rename it to CRSF specific :)

**Comment by AlessandroAU** [comment]
no this is something else. @cruwaller happy to merge?

**Review by cruwaller — APPROVED** [review]
This PR is hard to follow because all master-dev commits are shown as a changes. Proper rebase need to be used in future PRs...
